pkg/framework/resource/resource.go

```diff
@@ -34,7 +35,24 @@ func New(config ResourceConfig) (*Resource, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 	if config.ApprClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.ApprClient must not be empty", config)
+		config.Logger.Log("level", "debug", "message", fmt.Sprintf("%T.ApprClient is empty", config))
+
+		config.Logger.Log("level", "debug", "message", fmt.Sprintf("using default for %T.ApprClient", config))
+
+		c := apprclient.Config{
+			Fs:     afero.NewOsFs(),
+			Logger: config.Logger,
+
+			Address:      "https://quay.io",
+			Organization: "giantswarm",
```

Comment:
Would be good to have those in the `ResourceConfig` or at least as constants.

Reply:
The whole `apprClient` is part of the `ResourceConfig` - I just create one here with values I defaulted sensibly. Adding constants fields weird because I am just using them once so far. 

How can I improve this ?

Reply:
I think constants are also good in order no to hide values down in the code, just my opinion.

Reply:
I will add constants later when I add more functionality here. Going with this for now.